tuf/api/metadata.py

```diff
@@ -719,33 +756,30 @@ def get_verification_result(
         """
         role = self.get_delegated_role(delegated_role)
 
-        signed = set()
-        unsigned = set()
+        signed = {}
+        unsigned = {}
 
         for keyid in role.keyids:
             try:
                 key = self.get_key(keyid)
             except ValueError:
-                unsigned.add(keyid)
```

Comment:
Interesting. This means that in the new VerificationResult there might be role keyids that are neither in signed nor unsigned. That's probably okay. But should we mention it somewhere. Maybe in the PR description?

Reply:
Yeah it is a change -- I think I mentioned it in the first commit message , can add to PR decription.

 I was also considering just raising at this point (because clearly the role is in a bad state)... but then that sounds like punishing the delegated role signing  for delegator metadata error -- an error that can still leave the delegating role capable of signing.